app/models/ems_refresh/parsers/azure.rb

```diff
@@ -0,0 +1,214 @@
+module EmsRefresh
+  module Parsers
+    class Azure < ManageIQ::Providers::CloudManager::RefreshParser
+      VALID_LOCATION = /\w+/i
+
+      def self.ems_inv_to_hashes(ems, options = nil)
+        new(ems, options).ems_inv_to_hashes
+      end
+
+      def initialize(ems, options = nil)
+        @ems        = ems
+        ems.connect
+        @vmm        = ::Azure::Armrest::VirtualMachineManager.new
+        @asm        = ::Azure::Armrest::AvailabilitySetManager.new
+        @options    = options || {}
+        @data       = {}
+        @data_index = {}
+      end
+
+      def ems_inv_to_hashes
+        log_header = "MIQ(#{self.class.name}.#{__method__}) Collecting data for EMS : [#{@ems.name}] id: [#{@ems.id}]"
+
+        $log.info("#{log_header}...")
+        get_series
+        get_availability_sets
+        get_vms
+        clean_up_extra_flavor_keys
+        $log.info("#{log_header}...Complete")
+
+        @data
+      end
+
+      private
+
+      def get_series
+        series = []
+        get_locations.each do |location|
+          begin
+            series << @vmm.series(location)
+          rescue RestClient::BadGateway
+            next
+          end
+        end
+        series = series.flatten!.uniq
+        process_collection(series, :flavors) { |s| parse_series(s) }
+      end
+
+      def get_availability_sets
+        a_zones = @asm.list
+        process_collection(a_zones, :availability_zones) { |az| parse_az(az) }
+      end
+
+      def get_vms
+        vms = @vmm.get_vms
+        process_collection(vms, :vms) { |instance| parse_vm(instance) }
+      end
+
+      def process_collection(collection, key)
+        @data[key] ||= []
+
+        return if collection.nil?
+
+        collection.each do |item|
+          uid, new_result = yield(item)
+          @data[key] << new_result
+          @data_index.store_path(key, uid, new_result)
+        end
+      end
+
+      def parse_series(s)
+        name = uid = s['name']
+        new_result = {
+          :type           => "SeriesAzure",
```

Comment:
I *really* don't like that we call a subclass of Flavor by a different name.  That is just going to be an immediate confusion.  Can we please rename the model to `FlavorAzure`?